src/tests/global_variables/env_variables_defaults.py

```diff
@@ -1,95 +1,105 @@
 from abc import ABC
-from dataclasses import dataclass, field
+from dataclasses import dataclass
 from distutils.util import strtobool
 from pathlib import Path
-from typing import Any, ClassVar, List
 
 import consts
-from assisted_test_infra.test_infra.utils import get_env, operators_utils
+from assisted_test_infra.test_infra.utils.env_var import EnvVar
 from consts import env_defaults, resources
+from triggers.env_trigger import DataPool
 
 
 @dataclass(frozen=True)
-class _EnvVariablesDefaults(ABC):
-    ssh_public_key: str = get_env("SSH_PUB_KEY")
-    remote_service_url: str = get_env("REMOTE_SERVICE_URL")
-    pull_secret: str = get_env("PULL_SECRET")
-    offline_token: str = get_env("OFFLINE_TOKEN")
-    openshift_version: str = ""
-    base_dns_domain: str = get_env("BASE_DOMAIN", env_defaults.DEFAULT_BASE_DNS_DOMAIN)
-    masters_count: int = int(get_env("MASTERS_COUNT", get_env("NUM_MASTERS", env_defaults.DEFAULT_NUMBER_OF_MASTERS)))
-    workers_count: int = int(get_env("WORKERS_COUNT", get_env("NUM_WORKERS", env_defaults.DEFAULT_WORKERS_COUNT)))
-    nodes_count: int = masters_count + workers_count
-    day2_workers_count: int = int(get_env("NUM_DAY2_WORKERS", env_defaults.DEFAULT_DAY2_WORKERS_COUNT))
-    vip_dhcp_allocation: bool = bool(
-        strtobool(get_env("VIP_DHCP_ALLOCATION", str(env_defaults.DEFAULT_VIP_DHCP_ALLOCATION)))
+class _EnvVariables(DataPool, ABC):
+    ssh_public_key: EnvVar = EnvVar(["SSH_PUB_KEY"])
+    remote_service_url: EnvVar = EnvVar(["REMOTE_SERVICE_URL"])
+    pull_secret: EnvVar = EnvVar(["PULL_SECRET"])
+    deploy_target: EnvVar = EnvVar(["DEPLOY_TARGET"])
+    offline_token: EnvVar = EnvVar(["OFFLINE_TOKEN"])
+    openshift_version: EnvVar = EnvVar(["OPENSHIFT_VERSION"], default=consts.OpenshiftVersion.VERSION_4_9.value)
```

Comment:
default not taken from the service?

Reply:
It does.
I'll probably had to add documentation here also.
When removing the default from the Makefile it cased some tests to fail. This is a temporary fix until we will remove that dependency.